### PR TITLE
Removing the search bar from the home page

### PR DESCRIPTION
**Pull request by rememberlenny**
# What 

Remove the search bar from the home page. Add props to determine when the search bar should appear in the navbar.

# How

Use a prop that is set based on the router. The router pathname determines a value that gets passed to navbar.

# Reference

Resolving issue #513

### CODE REVIEW

File: ui/components/app-wrapper.js [hunk 1]
```diff
@@ -22,10 +22,17 @@ Router.onRouteChangeStart = NProgress.start;
 export default function wrapPage(Page, dataFn) {
   function WrappedPage(props) {
     if (props.err) return <ErrorPage err={props.err} />;
-    return <HeaderFooter><Page {...props} /></HeaderFooter>;
+    return <HeaderFooter showSearch={ props.url.pathname !== "/" }><Page {...props} /></HeaderFooter>;
```
Comment (cmc333333) [inline ui/components/app-wrapper.js]: Do you think we can get by without testing for global state? It's difficult to test and tends to be fragile.

Maybe we could change the function to `wrapPage(Page, dataFn, headerFooterProps)` and then have `Homepage` pass in `{ showSearch: false }`?

File: ui/components/navbar.js [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
 import SearchContainer from './search/search';
 
 
-function Navbar() {
+export default function Navbar({ showSearch }) {
+  let searchContainer;
+  if (showSearch) {
+    searchContainer = <SearchContainer />;
+  }
+
```
Comment (cmc333333) [inline ui/components/navbar.js]: Now that there's some logic in this function, should we add a test?

File: ui/components/navbar.js [hunk 3]
```diff
@@ -18,10 +25,12 @@ function Navbar() {
...
+        { searchContainer }
       </div>
     </div>
   );
 }
 
-export default Navbar;
+Navbar.propTypes = {
+  showSearch: PropTypes.bool
+};
```
Comment (cmc333333) [inline ui/components/navbar.js]: Should `showSearch` have a default of `true`?

File: ui/components/header-footer.js [hunk 4]
```diff
@@ -91,5 +91,6 @@ export default function HeaderFooter({ children }) {
 }
 
 HeaderFooter.propTypes = {
+  showSearch: PropTypes.bool,
```
Comment (cmc333333) [inline ui/components/header-footer.js]: Heads up, we might get a linting error here (once linting is part of CI again) due to not having a default.

File: ui/components/navbar.js (hunk 2 above)
Comment (rememberlenny) [reply ui/components/navbar.js]: Yep! Just added some.

File: ui/components/app-wrapper.js (hunk 1 above)
Comment (rememberlenny) [reply ui/components/app-wrapper.js]: I agree. I like this. Making an update.

File: ui/components/header-footer.js (hunk 4 above)
Comment (rememberlenny) [reply ui/components/header-footer.js]: Copy!

File: ui/components/navbar.js (hunk 3 above)
Comment (rememberlenny) [reply ui/components/navbar.js]: @cmc333333 Added the default value!

File: ui/__tests__/components/navbar.test.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+describe('<Navbar />', () => {
+  it('without a search bar when showSearch is false', () => {
+    const result = shallow(<Navbar showSearch={ false }/>);
+    expect(result.find(SearchContainer).length).toEqual(0);
+  });
+
+  it('show search bar when showSearch is true', () => {
+    const result = shallow(<Navbar showSearch={ true }/>);
+    expect(result.find(SearchContainer).length).toEqual(1);
```
Comment (cmc333333) [inline ui/__tests__/components/navbar.test.js]: Super minor, but `expect(...).toHaveLength(1)` will give us slightly better error messages.

File: ui/components/app-wrapper.js [hunk 6]
```diff
@@ -19,13 +19,20 @@ Router.onRouteChangeError = NProgress.done;
 Router.onRouteChangeStart = NProgress.start;
 
 
-export default function wrapPage(Page, dataFn) {
+export default function wrapPage(Page, dataFn, headerFooterParams = { showSearch: true }) {
```
Comment (cmc333333) [inline ui/components/app-wrapper.js]: Unfortunately, default values are shared across all calls to `wrapPage`. Combine this with a mutable value and we'll get very unexpected results.

We can get around that with something like:
```js
export default function wrapFunction(Page, dataFn, headerFooterParams) {
  const hfParams = headerFooterParams || { showSearch: true };
```

File: ui/components/app-wrapper.js [hunk 7]
```diff
@@ -19,13 +19,20 @@ Router.onRouteChangeError = NProgress.done;
...
   function WrappedPage(props) {
     if (props.err) return <ErrorPage err={props.err} />;
-    return <HeaderFooter><Page {...props} /></HeaderFooter>;
+    return <HeaderFooter showSearch={ headerFooterParams.showSearch }><Page {...props} /></HeaderFooter>;
```
Comment (cmc333333) [inline ui/components/app-wrapper.js]: Would it make sense to just pass any parameter the user requested? This way we won't have to modify this function should we add more parameters. Something like:
```js
return <HeaderFooter {...headerFooterParams}><Page {...props} /></HeaderFooter>;
```

Comment (rememberlenny) [reply ui/components/app-wrapper.js]: I had to look up how the spread syntax worked. This is brilliant.

**Comment by rememberlenny** [comment]
@cmc333333 Updates are up.

**Comment by cmc333333** [comment]
Thanks for accommodating my nits, @rememberlenny !

**Pull request merged**